sxt/execution/device/for_each.cc

```diff
@@ -30,12 +30,11 @@ namespace sxt::xendv {
 // concurrent_for_each
 //--------------------------------------------------------------------------------------------------
 xena::future<>
-concurrent_for_each(basit::index_range rng,
+concurrent_for_each(basit::index_range_iterator first, basit::index_range_iterator last,
```

Comment:
Correct me if I'm wrong - this basically allows you to define a custom index_range_iterator whose step isn't bound to the number of cuda devices, right? But no matter what, `last` has to be the final value of the index_range_iterator once it's exhausted?

Would it make more sense for the parameters to just be an `index_range` and a step value to pass to `basit::split`, then the other definition passes in `basdv::get_num_devices()` as the step value? It seems like the `last` parameter would make this error if defined wrong. Or am I missing something?

Reply:
Before calling the function, the code I'm working on needs to allocate some memory that's dependent on the distance (last - first); (ie the number of chunks). With this version, it has the first and last readily available to figure out the chunk count.

Also, passing in iterators is pretty conventional for C++ functions, see for example standard library functions: https://en.cppreference.com/w/cpp/algorithm/find

Reply:
Makes sense. Thanks for the info